cmd/integration-test/pkg/tests/workload_proxy.go

```diff
@@ -94,11 +95,18 @@ func AssertWorkloadProxy(testCtx context.Context, apiClient *client.Client, clus
 			svcURL = r.TypedSpec().Value.Url
 		})
 
+		defaultTransport, ok := http.DefaultTransport.(*http.Transport)
```

Comment:
I would suggest to use https://github.com/hashicorp/go-cleanhttp instead when you want to build a transport.

You can build a pooled transport as well, just make sure to CloseIdleConnections when done

Reply:
I hate `DefaultTransport`, as if something touches it somewhere, it might have wild effects on other parts of the code which are hard to track.

Reply:
Yep, I would never modify it - clone it, then modify that one. But cleanhttp indeed seems cleaner.

Reply:
I even mean we import e.g. some library (bad one), and it modifies `DefaultTransport` as a side-effect

Reply:
Oh right, then the clone will have that modification as well.